lib/pry-doc/pry_ext/show_source_with_c_internals/code_fetcher.rb

```diff
@@ -0,0 +1,116 @@
+require 'fileutils'
+require_relative 'c_file'
+require_relative 'symbol_extractor'
+
+module Pry::CInternals
+  class CodeFetcher
+    include Pry::Helpers::Text
+
+    class << self
+      attr_accessor :ruby_source_folder
+    end
+
+    # normalized
+    def self.ruby_version() RUBY_VERSION.tr(".", "_") end
+
+    self.ruby_source_folder = File.join(File.expand_path("~/.pry.d"), "ruby-#{ruby_version}")
+
+    attr_reader :line_number_style
+    attr_reader :symbol_extractor
+
+    def initialize(line_number_style: nil)
+      @line_number_style = line_number_style
+      @symbol_extractor = SymbolExtractor.new(self.class.ruby_source_folder)
+    end
+
+    def fetch_all_definitions(symbol)
+      infos = self.class.symbol_map[symbol]
+      return unless infos
+
+      result = ""
+      infos.count.times do |index|
+        result << fetch_first_definition(symbol, index).first << "\n"
+      end
+
+      return [result, infos.first.file]
+    end
+
+    def fetch_first_definition(symbol, index=nil)
+      infos = self.class.symbol_map[symbol]
+      return unless infos
+
+      info = infos[index || 0]
+      code = symbol_extractor.extract(info)
+
+      result = ""
+      result << "\n#{bold('From: ')}#{info.file} @ line #{info.line}:\n"
+      result << "#{bold('Number of implementations:')} #{infos.count}\n" unless index
+      result << "#{bold('Number of lines: ')} #{code.lines.count}\n\n"
+      result << Pry::Code.new(code, start_line_for(info.line), :c).
+                  with_line_numbers(use_line_numbers?).highlighted
+
+      return [result, info.file]
+
+    end
+
+    private
+
+    def use_line_numbers?
+      !!line_number_style
+    end
+
+    def start_line_for(line)
+      line_number_style == :'base-one' ? 1 : line || 1
+    end
+
+    def self.symbol_map
```

Comment:
The `private` keyword has no effect on this definition (these methods are still public). You should be using `private_class_method` instead.

Reply:
@kyrylo i know i wasn't actually trying to make them private though :)